src/common/warning.rs

```diff
@@ -3,14 +3,18 @@ use std::fmt;
 use crate::common::*;
 
 #[derive(Clone, Debug, PartialEq)]
-pub struct Warning {
+pub struct Warning<'l> {
     pub check_name: String,
-    line: LineEntry,
+    line: &'l LineEntry,
     message: String,
 }
```

Comment:
What do you think of a complete change in approach? 🤔 

As I see you have successfully abandoned `LineEntry` in `Fixers`, it seems to me that nothing prevents you from completely removing them from `Warnings`.

Where we need it, we will take the line by index, and at the same time we will completely solve the problem without lighttime or `Rc`.

Also, we can avoid the mentioned side effect with the output of warnings.
We will not need to modify them in the `HashMap`, we will be able to print them after the fixers work.

```suggestion
pub struct Warning {
    pub check_name: String,
    line_num: usize,
    message: String,
}
```

Reply:
I like this idea 👍

Reply:
@atsmtat What do you think about this?

Reply:
I like this idea too 👍 In fact, it crossed my mind before, but wasn't sure if you'll like it as it's a major change. Since we all like it, I gotta try it out 🙂